FSCP-4: Form screen

sergey-aslonyns opened the pull request.

Commit:
переезд

Commit:
created form screen

tplaymeow commented on .DS_Store:
Добавь в gitignore

tplaymeow commented on fablescope/fablescope/Extensions/UIColor+.swift:
```diff
@@ -10,4 +10,6 @@ import UIKit
 extension UIColor {
     static let fableGreen = UIColor(named: "fable_green")!
+    static let frostColor = UIColor(named: "frost_color")!
+    
```
Удали пустую строку

tplaymeow commented on fablescope/fablescope/Extensions/UIImage+.swift:
```diff
@@ -13,4 +13,7 @@ extension UIImage {
     static let mainScreenBack = UIImage(named: "main_screen_background")!
+    static let witchHat = UIImage(named: "witch_hat")!
+    static let crown = UIImage(named: "crown")!
+    
```
Удали пустую строку

tplaymeow commented on fablescope/fablescope/Form Screen/CollectionViewCell.swift:
```diff
@@ -0,0 +1,64 @@
+class CollectionViewCell: UICollectionViewCell {
+    var data: CastomData? {
+        didSet{
+            guard let data = data else { return }
```
Можно просто guard let data else { return }

tplaymeow commented on fablescope/fablescope/Form Screen/CollectionViewCell.swift:
```diff
@@ -0,0 +1,64 @@
+
+    var isSelectedCell: Bool = false {
+        didSet {
+            if isSelectedCell == false {
```
if !isSelectedCell

tplaymeow commented on fablescope/fablescope/Form Screen/CollectionViewCell.swift:
```diff
@@ -0,0 +1,64 @@
+                cellImage.tintColor = .frostColor
+                backgroundColor = .white
+                layer.borderWidth = 3
+                layer.borderColor = UIColor.fableGreen.cgColor
```
Почему в другой ветке не убираешь это?

tplaymeow commented on fablescope/fablescope/Form Screen/CollectionViewCell.swift:
```diff
@@ -0,0 +1,64 @@
+    }
+
+    func setupConstraints() {
+            NSLayoutConstraint.activate([
```
Отступов дофига

tplaymeow commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,132 @@
+class FromScreenViewController: UIViewController {
+    private let createButton = ReadyButton()
+    let data = [
+        CastomData( image: .witchHat ),
```
Пробелы перед лейблом у аргумента не нужен

tplaymeow commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,132 @@
+            collectionView.bottomAnchor.constraint(equalTo: view.bottomAnchor),
+            collectionView.leftAnchor.constraint(equalTo: view.leftAnchor),
+            collectionView.rightAnchor.constraint(equalTo: view.rightAnchor)
+            ])
```
Неправильный отступ

tplaymeow commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,132 @@
+            cell?.isSelectedCell = true
+        } else {
+            cell?.isSelectedCell = false
+        }
```
Слишком сложная конструкция
Модно просто a = !a
Или a.toggle()

tplaymeow commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,132 @@
+
+extension FromScreenViewController: UICollectionViewDelegateFlowLayout {
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        return CGSize(width: collectionView.frame.width/3.75, height: collectionView.frame.width/3.75)
```
Что такое 3.75? В коде не должно быть magic numbers

Mov4D replied:
Да вынеси в const и дай логичное название

tplaymeow commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,132 @@
+
+extension FromScreenViewController: UICollectionViewDelegateFlowLayout {
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        return CGSize(width: collectionView.frame.width/3.75, height: collectionView.frame.width/3.75)
```
Перед и после операторов нужен пробел

tplaymeow commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,132 @@
+    }
+
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, insetForSectionAt section: Int) -> UIEdgeInsets {
+           return UIEdgeInsets(top: 20, left: 20, bottom: 20, right: 20)
```
Выравнивания опять сделали брррр. Нажимай ты cmd-i

rofle100lvl replied:
пожалуйста

tplaymeow commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,132 @@
+        static let createButtonHeight: CGFloat = 59
+        static let createButtonBottomInset: CGFloat = -40
+    }
+
```
Удали строку

tplaymeow commented on fablescope/fablescope/Form Screen/ReadyButton.swift:
```diff
@@ -0,0 +1,51 @@
+
+import UIKit
+final class ReadyButton: UIControl {
```
Добавь пробел после import

tplaymeow commented on fablescope/fablescope/Main Screen/MainScreenViewController.swift:
```diff
@@ -93,4 +98,5 @@ extension MainScreenViewController {
         static let autoButtonHeigh: CGFloat = 166
         static let buttonStackViewSpacing: CGFloat = 20
     }
+
```
Удали пустую линию

tplaymeow commented on fablescope/fablescope/Main Screen/MainScreenViewController.swift:
```diff
@@ -81,6 +81,11 @@ private extension MainScreenViewController {
+        navigationController?.pushViewController(FromScreenViewController(), animated: true)
     }
 }
 
```
![image](https://github.com/rep-co/fablescope-ios/assets/35903356/87a5be95-9e5d-415b-80ad-ec42f03f449e)

Mov4D commented on fablescope/fablescope/Form Screen/CollectionViewCell.swift:
```diff
@@ -0,0 +1,64 @@
+ import UIKit
```
лишний пробел перед import

Mov4D commented on fablescope/fablescope/Form Screen/CollectionViewCell.swift:
```diff
@@ -0,0 +1,64 @@
+        }
+    }
+
+    var isSelectedCell: Bool = false {
```
все что внутри didSet нужно в отдельную функцию вынести

Mov4D commented on fablescope/fablescope/Form Screen/CollectionViewCell.swift:
```diff
@@ -0,0 +1,64 @@
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
```
Добавляй комментарии к extension // MARK: - Что ты делаешь внутри

Mov4D commented on fablescope/fablescope/Form Screen/CollectionViewCell.swift:
```diff
@@ -0,0 +1,64 @@
+
+private extension CollectionViewCell {
+    func setupUI() {
+        addSubwiews()
+        setupConstraints()
```
Их лучше в init вызывать

Mov4D commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,132 @@
+
+import UIKit
+
+struct CastomData {
```
Плохое название + не правильно написано

Mov4D commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,132 @@
+}
+
+class FromScreenViewController: UIViewController {
+    private let createButton = ReadyButton()
```
почему экземпляр create а класс Ready?

Mov4D commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,132 @@
+    var image: UIImage
+}
+
+class FromScreenViewController: UIViewController {
```
final

Mov4D commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,132 @@
+    ]
+    private let collectionView: UICollectionView = {
+        let layout = UICollectionViewFlowLayout()
+        let cv = UICollectionView(frame: .zero, collectionViewLayout: layout)
```
двумя буквами не называем переменные

Mov4D commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,132 @@
+        let layout = UICollectionViewFlowLayout()
+        let cv = UICollectionView(frame: .zero, collectionViewLayout: layout)
+        cv.translatesAutoresizingMaskIntoConstraints = false
+        cv.register(CollectionViewCell.self, forCellWithReuseIdentifier: "cell")
```
регистрацию ячейки лучше в отдельный метод вынести

Mov4D commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,132 @@
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupUI()
+        title = "Выбери все что хочешь"
```
в SetupUI вынеси

Mov4D commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,132 @@
+
+private extension FromScreenViewController{
+    func setupUI() {
+        addSubwiews()
```
вызов должен быть из viewDidLoad

Mov4D commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,132 @@
+        collectionView.delegate = self
+        collectionView.dataSource = self
+        view.addSubview(collectionView)
+        collectionView.backgroundColor = .white
```
это явно не в addSubviews должно быть

Mov4D commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,132 @@
+    }
+
+    func addSubwiews() {
+        collectionView.delegate = self
+        collectionView.dataSource = self
```
и это

Mov4D commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,132 @@
+            .foregroundColor: UIColor.black,
+            .font: UIFont.pacificoRegular
+        ]
+        setupConstraints()
```
вызывай из viewDidLoad

Mov4D commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,132 @@
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "cell", for: indexPath) as! CollectionViewCell
```
forseUnwrap

Mov4D commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,132 @@
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "cell", for: indexPath) as! CollectionViewCell
+        cell.data = self.data[indexPath.row]
+        cell.backgroundColor = .fableGreen
```
ячейку внутри своего класса настраивай

Mov4D commented on fablescope/fablescope/Form Screen/ReadyButton.swift:
```diff
@@ -0,0 +1,51 @@
+    func addSubwiews() {
+        addSubview(label)
+        label.translatesAutoresizingMaskIntoConstraints = false
+        setupReadyButton()
+        setupConstraints()
+        setupLabel()
```
вызов этих функций должен быть явно не здесь

Mov4D commented on fablescope/fablescope/Main Screen/MainScreenViewController.swift:
```diff
@@ -81,6 +81,11 @@ private extension MainScreenViewController {
+        createButton.addTarget(self, action: #selector(onClick), for: .touchUpInside)
+    }
+
+    @objc func onClick() {
```
плохое название не понятно на что именно нажали лучше didTapCreateButton

Mov4D commented on fablescope/fablescope/Main Screen/MainScreenViewController.swift:
```diff
@@ -81,6 +81,11 @@ private extension MainScreenViewController {
+    }
+
+    @objc func onClick() {
+        navigationController?.pushViewController(FromScreenViewController(), animated: true)
```
Навигация должна быть через Router, посмотри MVC + Router

Mov4D commented:
грязный код

rofle100lvl commented on fablescope/fablescope/Form Screen/CollectionViewCell.swift:
```diff
@@ -0,0 +1,64 @@
+ import UIKit
+
+class CollectionViewCell: UICollectionViewCell {
+    var data: CastomData? {
```
CustomData*

rofle100lvl commented on fablescope/fablescope/Form Screen/ReadyButton.swift:
```diff
@@ -0,0 +1,51 @@
+        NSLayoutConstraint.activate([
+            label.bottomAnchor.constraint(equalTo: self.bottomAnchor),
+            label.topAnchor.constraint(equalTo: self.topAnchor),
+            label.rightAnchor.constraint(equalTo: self.rightAnchor),
```
![image](https://github.com/rep-co/fablescope-ios/assets/45801227/5fc3086d-573b-465a-ae8e-bc7e4ad1549e)

rofle100lvl commented:
<img width="135" alt="Screenshot 2023-12-12 at 16 30 36" src="https://github.com/rep-co/fablescope-ios/assets/45801227/3c63583a-9082-406c-bc9b-ecf873832d54">
Давай без русского языка в коммитах

tplaymeow commented on fablescope/fablescope/Form Screen/CollectionViewCell.swift:
```diff
@@ -0,0 +1,64 @@
+ import UIKit
+
+class CollectionViewCell: UICollectionViewCell {
```
CollectionViewCell - странное название. Название класса должно отражать его назначение. Это же не просто какая-то случайная ячейка, это ячейка, которая отражает конкретные данные

tplaymeow replied:
Сделай класс final

tplaymeow commented on fablescope/fablescope/AppDelegate.swift:
```diff
@@ -18,7 +18,7 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
         // Override point for customization after application launch
```
Так и не исправил комментарий, который я просил убрать еще в прошлом пр-е. УДОЛИ ЕГО!!!

tplaymeow commented on fablescope/fablescope/Extensions/UIColor+.swift:
```diff
@@ -10,4 +10,6 @@ import UIKit
 
 extension UIColor {
     static let fableGreen = UIColor(named: "fable_green")!
+    static let frostColor = UIColor(named: "frost_color")!
```
Почему именно такие названия у цветов?

tplaymeow commented on fablescope/fablescope/Form Screen/CollectionViewCell.swift:
```diff
@@ -0,0 +1,64 @@
+
+class CollectionViewCell: UICollectionViewCell {
+    var data: CastomData? {
+        didSet{
```
добавь пробел перед фигурной скобкой

tplaymeow commented on fablescope/fablescope/Form Screen/CollectionViewCell.swift:
```diff
@@ -0,0 +1,64 @@
+    }
+
+    private let  cellImage: UIImageView = {
+       let iv = UIImageView()
```
iv - плохое название, напиши imageView

tplaymeow commented on fablescope/fablescope/Form Screen/CollectionViewCell.swift:
```diff
@@ -0,0 +1,64 @@
+        setupUI()
+    }
+
+    required init?(coder : NSCoder) {
```
Удали пробел перед двоеточием

Commit:
corrected remarks except gitignore

Mov4D commented on fablescope/fablescope/Extensions/UIImage+.swift:
```diff
@@ -1,19 +1,9 @@
-
 import UIKit
 
-// swiftlint:disable force_unwrap
```
Зачем эту строчку убрал?

Mov4D commented on fablescope/fablescope/Form Screen/CollectionViewCell.swift:
```diff
@@ -21,44 +21,47 @@ class CollectionViewCell: UICollectionViewCell {
     }
-
+    
     private let  cellImage: UIImageView = {
```
2 пробела

Mov4D commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -1,36 +1,36 @@
 import UIKit
 
-struct CastomData {
+struct ViewModel {
```
структура ViewModel не должна лежать просто в воздухе

Mov4D commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -127,6 +126,6 @@ extension FromScreenViewController {
         static let createButtonRightInset: CGFloat = -20
         static let createButtonHeight: CGFloat = 59
         static let createButtonBottomInset: CGFloat = -40
+        static let collectionViewrRatio: CGFloat = 3.75
```
неверно написано

tplaymeow commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,131 @@
+final class FromScreenViewController: UIViewController {
+    private let createButton = ReadyButton()
+    let data = [
+        ViewModel( image: .witchHat ),
```
Убери лишние пробелы плиз

Commit:
Have any changes been made

tplaymeow commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,133 @@
+    var image: UIImage
+}
+
+final class FromScreenViewController: UIViewController {
```
Так "FormScreen" или "FromScreen"?

tplaymeow commented on fablescope/fablescope/Form Screen/CollectionViewCell.swift:
```diff
@@ -0,0 +1,67 @@
+        backgroundColor = .fableGreen
+    }
+    
+    func addSubwiews() {
```
Очепятка addSubviews

Commit:
comments have been corrected except .gitignore

tplaymeow reviewed (approved):
Круто! Отдай в тестирование

sergey-aslonyns committed:
Premium
Украинский язык теперь доступен в Reverso.
Присоединяйтесь к Reverso, это удобно и бесплатно!Зарегистрироваться Войти
fix a bug

Спрягать этот глагол
fix a bug

Premium
Украинский язык теперь доступен в Reverso.
Присоединяйтесь к Reverso, это удобно и бесплатно!Зарегистрироваться Войти
fix a bug

Спрягать этот глагол
fix a bug

fix a bag

Mov4D commented on fablescope/fablescope/Fairy Tale Screen/FairyTaleScreenController.swift:
```diff
@@ -0,0 +1,9 @@
+import UIKit
+class FairyTaleScreenController: UIViewController {
```
```suggestion
import UIKit

final class FairyTaleScreenController: UIViewController {
```

Mov4D commented on fablescope/fablescope/Fairy Tale Screen/FairyTaleScreenController.swift:
```diff
@@ -0,0 +1,9 @@
+import UIKit
+class FairyTaleScreenController: UIViewController {
+    let skrollView = UIScrollView()
+    let  scrollView1 = UIView()
```
неверный уровень доступа

Mov4D commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,131 @@
+final class FormScreenViewController: UIViewController {
+    private let submitButton = SubmitButton()
+    
+    var data = [
```
почему не private?

Mov4D commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,131 @@
+        CollectionViewCell.ViewModel(image: .witchHat, isSelected: false),
+        CollectionViewCell.ViewModel(image: .crown, isSelected: false),
+        CollectionViewCell.ViewModel(image: .magicWandIcon, isSelected: false)
+    ]
```
```suggestion
    ]
    
```

Mov4D commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,131 @@
+    }
+}
+
+private extension FormScreenViewController{
```
```suggestion
private extension FormScreenViewController {
```

Mov4D commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,131 @@
+
+private extension FormScreenViewController{
+
+    func setupCollectionView(){
```
```suggestion
    func setupCollectionView() {
```

Mov4D commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,131 @@
+    }
+
+    func setupUI() {
+        title = "Выбери все что хочешь"
```
в конст вынеси

Mov4D commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,131 @@
+
+    func addSubviews() {
+        view.addSubview(collectionView)
+        collectionView.backgroundColor = .white
```
почему это в addSubviews?

Mov4D commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,131 @@
+        view.addSubview(collectionView)
+        collectionView.backgroundColor = .white
+        view.addSubview(submitButton)
+        submitButton.translatesAutoresizingMaskIntoConstraints = false
+        navigationController?.navigationBar.titleTextAttributes = [
```
почему это в addSubviews?

Mov4D commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,131 @@
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "cell", for: indexPath) as! CollectionViewCell
```
избавься от !

Mov4D commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,131 @@
+    }
+
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        data[indexPath.row] = .init(image: data[indexPath.row].image, isSelected:  !data[indexPath.row].isSelected)
```
Что если по этому индексу не будет элемента?

Mov4D commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,131 @@
+
+extension FormScreenViewController: UICollectionViewDelegateFlowLayout {
+
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        return CGSize(width: collectionView.frame.width/Const.collectionViewRatio, height: collectionView.frame.width/Const.collectionViewRatio)
+    }
```
Сделай норм переносы

Mov4D commented on fablescope/fablescope/Form Screen/SubmitButton.swift:
```diff
@@ -0,0 +1,52 @@
+import UIKit
+
+final class SubmitButton: UIControl {
+    private let label = UILabel()
```
плохое название

sergey-aslonyns committed:
bug fixes

rofle100lvl commented on fablescope/fablescope/Fairy Tale Screen/FairyTaleScreenController.swift:
```diff
@@ -0,0 +1,10 @@
+import UIKit
+
+class FairyTaleScreenController: UIViewController {
+    private let skrollView = UIScrollView()
```
scroll

rofle100lvl commented on fablescope/fablescope/Fairy Tale Screen/FairyTaleScreenController.swift:
```diff
@@ -0,0 +1,10 @@
+
+class FairyTaleScreenController: UIViewController {
+    private let skrollView = UIScrollView()
+    private let  scrollView1 = UIView()
```
название scrollView1 ни о чем не говорит, и почему в переменной scrollView лежит UIView? + пробел лишний

rofle100lvl commented on fablescope/fablescope/Form Screen/CollectionViewCell.swift:
```diff
@@ -0,0 +1,80 @@
+import UIKit
+
+final class CollectionViewCell: UICollectionViewCell {
+
```
строка лишняя

rofle100lvl commented on fablescope/fablescope/Form Screen/CollectionViewCell.swift:
```diff
@@ -0,0 +1,80 @@
+
+    var isSelectedCell: Bool = false {
+        didSet {
+            setupDidSet()
```
название непонятное  переименуй во что-то типа onCellSelection или что-нибудь такое

rofle100lvl commented on fablescope/fablescope/Form Screen/CollectionViewCell.swift:
```diff
@@ -0,0 +1,80 @@
+        }
+    }
+
+    private let  cellImage: UIImageView = {
```
пробел лишний

rofle100lvl commented on fablescope/fablescope/Form Screen/CollectionViewCell.swift:
```diff
@@ -0,0 +1,80 @@
+// MARK: - setup view
+
+private extension CollectionViewCell {
+
```
строка лишняя

rofle100lvl commented on fablescope/fablescope/Form Screen/CollectionViewCell.swift:
```diff
@@ -0,0 +1,80 @@
+        contentView.addSubview(cellImage)
+    }
+
+    func setupConstraints() {
```
лайк, что методы повыносил

rofle100lvl commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,136 @@
+}
+
+private extension FormScreenViewController {
+
```
строка лишняя

rofle100lvl commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,136 @@
+}
+
+extension FormScreenViewController: UICollectionViewDataSource, UICollectionViewDelegate {
+
```
строка лишняя

rofle100lvl commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,136 @@
+}
+
+extension FormScreenViewController: UICollectionViewDelegateFlowLayout {
+
```
строка лишняя

rofle100lvl commented on fablescope/fablescope/Form Screen/FormScreenViewController.swift:
```diff
@@ -0,0 +1,136 @@
+}
+
+extension FormScreenViewController {
+
```
тут тоже

rofle100lvl commented on fablescope/fablescope/Form Screen/SubmitButton.swift:
```diff
@@ -0,0 +1,52 @@
+}
+
+private extension SubmitButton {
+
```
и тут

rofle100lvl commented:
PR становится слишком большим, ничего нового сюда больше не добавляй. Доисправляй комменты и вливаем

sergey-aslonyns committed:
fix pr

rofle100lvl commented:
жоска ты меня отшил
